src/gt4py/next/iterator/ir_utils/common_pattern_matcher.py

```diff
@@ -7,13 +7,52 @@
 # SPDX-License-Identifier: BSD-3-Clause
 
 from collections.abc import Iterable
-from typing import Any, TypeGuard
+from typing import Any, Generic, List, TypeAlias, TypeGuard, TypeVar
 
 from gt4py.next.iterator import ir as itir
 from gt4py.next.iterator.ir_utils import ir_makers as im
 
 
-def is_applied_lift(arg: itir.Node) -> TypeGuard[itir.FunCall]:
+_Fun = TypeVar("_Fun", bound=itir.Expr)
+
+
+class _IsCall(itir.FunCall, Generic[_Fun]):
+    fun: _Fun
+    args: List[itir.Expr]
+
+
+_IsCallToPattern: TypeAlias = _IsCall[itir.SymRef]
+
+
+def is_call_to(node: Any, fun: str | Iterable[str]) -> TypeGuard[_IsCallToPattern]:
+    """
+    Match call expression to a given function.
+
+    If the `node` argument is not an `itir.Node` the function does not error, but just returns
+    `False`. This is useful in visitors, where sometimes we pass a list of nodes or a leaf
+    attribute which can be anything.
+
+    >>> from gt4py.next.iterator.ir_utils import ir_makers as im
+    >>> node = im.plus(1, 2)
+    >>> is_call_to(node, "plus")
+    True
+    >>> is_call_to(node, "minus")
+    False
+    >>> is_call_to(node, ("plus", "minus"))
+    True
+    """
+    assert not isinstance(fun, itir.Node)  # to avoid accidentally passing the fun as first argument
+    if isinstance(fun, (list, tuple, set, Iterable)) and not isinstance(fun, str):
+        return any((is_call_to(node, f) for f in fun))
+    return (
+        isinstance(node, itir.FunCall) and isinstance(node.fun, itir.SymRef) and node.fun.id == fun
+    )
```

Comment:
Shouldn't be simpler to change the order of the conditions here?
```suggestion
    if isinstance(fun, str):
        return (
            isinstance(node, itir.FunCall) and isinstance(node.fun, itir.SymRef) and node.fun.id == fun
        )
    else: 
        return any((is_call_to(node, f) for f in fun))
```